### PR TITLE
Improve error message when user enters wrong command

### LINKED ISSUE #147: add command does not work
**Describe the bug**
under some circumstances, add command fails to work

**To Reproduce**
1. type add

**Expected behavior**
help needs to be shown

**Screenshots**
image

**Additional context**

<hr>

**Reported by:** @alexchong711
**Severity:** Low

<sub>[original: nus-cs2103-AY1819S2/pe-dry-run#408]</sub>

### LINKED ISSUE #149: find command not found
**Describe the bug**
find command not found

**To Reproduce**
1. type googlenews JpKio9*((*23
2. type find

**Expected behavior**
help should be shown

**Screenshots**
image

**Additional context**

<hr>

**Reported by:** @alexchong711
**Severity:** Low

<sub>[original: nus-cs2103-AY1819S2/pe-dry-run#481]</sub>

### LINKED ISSUE #161: The context is not obviously enough displayed
Responsible for:
* #139
* #140 
* #141
* #144 
* #145 
* #147
* #149

### PR DESCRIPTION
**Pull request by thomastanck**
Resolves #149, #147, #145, #144, #141, #140, #139.
Resolves #161 for now.

### Summary of changes:
- Change error message of unknown command to:
```
Command not found in current context: %s.
Please check that you have entered your command correctly or if you are in the right context
```

### CODE REVIEW

**Comment by qjqqyy** [comment]
Making this PR resolve #161 for now. The remaining UI improvements are minor and thus deferred to v2.0 or later.

File: src/main/java/seedu/address/logic/parser/EntryBookParser.java [hunk 1]
```diff
@@ -18,12 +18,13 @@
...
 import seedu.address.logic.commands.SelectCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.ModelContext;
 
 /**
  * Represents the base parser for any context.
  * It successfully parses a command if and only if the command is context-switching, exit, help or history.
  */
-public class EntryBookParser {
+public abstract class EntryBookParser {
```
Comment (qjqqyy) [inline src/main/java/seedu/address/logic/parser/EntryBookParser.java:27]: good catch! we don't actually want to instantiate this parser!

**Review by qjqqyy — APPROVED** [review]
looksgoodtomeeee

File: src/main/java/seedu/address/logic/parser/EntryBookParser.java (hunk 1 above)
Comment (thomastanck) [reply src/main/java/seedu/address/logic/parser/EntryBookParser.java:27]: Thanks!

**Pull request merged**